src/renderer/canvas/canvas-renderer.ts

```diff
@@ -98,6 +98,10 @@ export class CanvasRenderer<N extends INodeBase, E extends IEdgeBase> extends Em
 
     // Clear drawing.
     this._context.clearRect(0, 0, this.width, this.height);
+    if (this._settings.backgroundColor) {
+      this._context.fillStyle = this._settings.backgroundColor.toString();
```

Comment:
Since the `backgroundColor` can be a string, do we validate anywhere if it's a valid color hex?

Or do we leave this to the HTML5 canvas API? 

I tried this on the canvas and nothing happens if an unexpected value is input. Should we maybe warn the user somewhere?

Reply:
Good note!

I've also tested and because it is not doing anything "wrong" on bad input, I left it as is because that is also how `canvas` works, meaning if it has something unhandled, it won't do it. The top-level usage, in our case the user of orb API, should handle the errors and validations of this matter. Same as how `canvas` is not breaking that we are using. Like a propagation. I would also skip warnings too in this case too because we are not doing any fallback action (like we do in the web-workers case).

What do you think?

Reply:
Yep, that makes sense.
But still, the canvas behavior itself is a bit strange. Usually, successful actions are silent.

Reply:
Agree :)